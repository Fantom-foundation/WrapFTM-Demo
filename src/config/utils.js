export function envToBool(value) {
    return value === 'true' || value === '1';
}
