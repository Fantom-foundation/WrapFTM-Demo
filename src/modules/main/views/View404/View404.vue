<script setup>
import { FCard } from 'fantom-vue3-components';
import Heading1 from '@/modules/common/components/Heading1/Heading1.vue';
</script>

<template>
    <div class="view404">
        <FCard class="view404_message">
            <Heading1 class="h3">{{ $t('main.view404.title') }}</Heading1>
            <p>{{ $t('main.view404.text') }}</p>
            <p>
                <RouterLink to="/">{{ $t('main.view404.linkTitle') }}</RouterLink>
            </p>
        </FCard>
    </div>
</template>

<style lang="scss">
.view404 {
    &_message {
        width: 90%;
        max-width: 600px;
        margin: var(--f-spacer-7) auto 0 auto;

        h1 {
            font-weight: bold;
            margin-top: 0;
        }
    }
}
</style>
