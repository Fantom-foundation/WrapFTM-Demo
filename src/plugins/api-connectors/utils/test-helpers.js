export class MyApiConnectorMock {
    __errors__ = {};
    __results__ = {};

    async processQuery() {
        return new Promise((resolve, reject) => {
            // console.log(this.__errors__.processQuery);
            if (this.__errors__.processQuery !== undefined) {
                reject(this.__errors__.processQuery);
            } else if (this.__results__.processQuery !== undefined) {
                resolve(this.__results__.processQuery);
            } else {
                resolve('result');
            }
        });
    }

    async processMutation() {
        return new Promise((resolve, reject) => {
            // console.log(this.__errors__.processQuery);
            if (this.__errors__.processMutation !== undefined) {
                reject(this.__errors__.processMutation);
            } else if (this.__results__.processMutation !== undefined) {
                resolve(this.__results__.processMutation);
            } else {
                resolve('result');
            }
        });
    }

    processResult(result) {
        return result;
    }
}
