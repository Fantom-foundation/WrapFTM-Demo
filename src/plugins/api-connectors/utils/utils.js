import { isProxy, isReactive, isRef, toRaw, unref } from 'vue';

/**
 * @param {function[]} functions
 * @param args
 */
export function callFunctions(functions, ...args) {
    functions.forEach((fn) => {
        if (typeof fn === 'function') {
            fn(...args);
        }
    });
}

export function deepToRaw(sourceObj) {
    const objectIterator = (input) => {
        if (Array.isArray(input)) {
            return input.map((item) => objectIterator(item));
        }
        if (isRef(input)) {
            return objectIterator(unref(input));
        }
        if (isReactive(input) || isProxy(input)) {
            return objectIterator(toRaw(input));
        }
        if (input && typeof input === 'object') {
            return Object.getOwnPropertyNames(input).reduce((acc, key) => {
                acc[key] = objectIterator(input[key]);
                return acc;
            }, {});
        }
        return input;
    };

    return objectIterator(sourceObj);
}
