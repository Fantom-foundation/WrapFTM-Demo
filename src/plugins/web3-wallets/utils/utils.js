import { toInt } from '@/utils/big-number/big-number.js';
import { CHAIN_NAMES } from '@/plugins/web3-wallets/utils/chain-names.js';

export function getChainNameById(chainId) {
    const chainIdI = toInt(chainId);

    return CHAIN_NAMES[chainIdI] || '';
}
