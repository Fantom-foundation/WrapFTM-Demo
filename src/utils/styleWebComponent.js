function applyStyles(elem, styles) {
    Object.keys(styles).forEach((prop) => {
        elem.style[prop] = styles[prop];
    });
}

function findWebComponent(wcSelector) {
    const wcs = wcSelector.split('>');
    let wc = null;

    wcs.forEach((wcS) => {
        wc = (wc === null ? document : wc?.shadowRoot)?.querySelector(wcS);
    });

    return wc;
}

/**
 * @param {string} wcSelector
 * @param {Object<string, Object>} styles
 */
export function styleWebComponent(wcSelector, styles) {
    const wc = findWebComponent(wcSelector);

    if (!wc) {
        throw new Error(`Can't find web component '${wcSelector}'`);
    }

    if (!wc.shadowRoot) {
        throw new Error(`'${wcSelector}' is not a web component`);
    }

    Object.keys(styles).forEach((selector) => {
        if (selector === 'root') {
            applyStyles(wc, styles[selector]);
        } else {
            const elems = wc.shadowRoot.querySelectorAll(selector);

            if (elems.length > 0) {
                elems.forEach((elem) => {
                    applyStyles(elem, styles[selector]);
                });
            } else {
                console.warn(`Can't find element(s) '${selector}'`);
            }
        }
    });
}
