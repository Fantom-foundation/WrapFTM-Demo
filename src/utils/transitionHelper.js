/**
 * Code taken from: https://developer.chrome.com/docs/web-platform/view-transitions/
 * Current view transitions api support: https://caniuse.com/?search=view%20transitions
 *
 * @param {boolean} skipTransition
 * @param {array} classNames
 * @param {function} updateDOM
 * @return {{updateCallbackDone: Promise<unknown>, ready: Promise<never>, skipTransition: skipTransition, finished: Promise<unknown>}|*}
 */
export function transitionHelper({ skipTransition = false, classNames = [], updateDOM }) {
    if (skipTransition || !document.startViewTransition) {
        const updateCallbackDone = Promise.resolve(updateDOM()).then(() => {});
        const ready = Promise.reject(Error('View transitions unsupported'));

        // Avoid spamming the console with this error unless the promise is used.
        ready.catch(() => {});

        return {
            ready,
            updateCallbackDone,
            finished: updateCallbackDone,
            skipTransition: () => {},
        };
    }

    document.documentElement.classList.add(...classNames);

    const transition = document.startViewTransition(updateDOM);

    transition.finished.finally(() =>
        document.documentElement.classList.remove(...classNames)
    );

    return transition;
}
