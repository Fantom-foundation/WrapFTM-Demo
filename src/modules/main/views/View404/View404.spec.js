import { mount } from '@vue/test-utils';
import { destroyWrapper } from 'fantom-vue3-components/src/test/utils.js';
import View404 from './View404.vue';

let wrapper = null;

function createWrapper(options = {}) {
    return mount(View404, options);
}

afterEach(() => {
    destroyWrapper(wrapper);
});

describe('View404', () => {
    it('should match snapshot', () => {
        wrapper = createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });
});
