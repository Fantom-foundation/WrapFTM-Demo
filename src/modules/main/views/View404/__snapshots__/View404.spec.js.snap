// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`View404 > should match snapshot 1`] = `
<div
  class="view404"
>
  <div
    class="fcard view404_message"
  >
    
    <h1
      class="h3"
      data-focus="true"
      tabindex="-1"
    >
      
      Page not found
      
    </h1>
    <p>
      Looks like you've followed a broken link or entered an URL that doesn't exist on this site.
    </p>
    <p>
      <a
        class=""
        href="/"
      >
        Back to home
      </a>
    </p>
    
  </div>
</div>
`;
